app/Http/Controllers/LocalizationController.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\App;
+
+class LocalizationController extends Controller
+{
+    public function index($locale)
+    {
+        App::setLocale($locale);
```

Comment:
Doublon avec le middleware

Reply:
Pas trop encore sûr sur les middlewares mais l'objectif du contrôller c'est de changer la langue avec une route et celui le middleware de récupérer/présever la langue ?
Je me suis basé sur https://www.larashout.com/how-to-create-multilingual-website-using-laravel-localization

Reply:
En dessus tu fait un `session()->put('locale', $locale);`. Le middleware fait un `App::setLocale(session()->get('locale'));`. Donc tu va set ta locale 2 fois

Reply:
(Puisque toutes les requêtes passent par le middleware)

Reply:
Le but du controller c'est de seulement stocker une locale dans la session. Le middleware va lui activer la locale en regardant la session.

Reply:
Donc on peut supprimer le middleware tel quel ?

Reply:
Non, `App::setLocale` n'est pas persistent de ce que je lis, donc il faut le faire à chaque requête (d'où le middleware). Ici le `App::setLocale` dans le controlleur est inutile puisque tu vas forcément passer par le middleware quand tu va rechercher la page, donc quand tu changes de locale tu exécute `App::setLocale` deux fois

Reply:
076e0de